userspace/libsinsp/sinsp_events_ppm_sc.cpp

```diff
@@ -0,0 +1,309 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include <sinsp_events.h>
+
+libsinsp::events::set<ppm_sc_code> libsinsp::events::enforce_sinsp_state_ppm_sc(libsinsp::events::set<ppm_sc_code> ppm_sc_of_interest)
+{
+	std::vector<uint32_t> minimum_syscalls(PPM_SC_MAX, 0);
+
+	/* Should never happen but just to be sure. */
+	if(scap_get_modifies_state_ppm_sc(minimum_syscalls.data()) != SCAP_SUCCESS)
```

Comment:
adding to @jasondellaluce point, given we refactor from the ground up, would love to see `scap_get_modifies_state_ppm_sc` be defined in this API and if you agree @FedeDP we might as well do the refactor of only enforcing the EF_MODIFIES_STATE and drop the UF_NEVER_DROP enforcement, but this can maybe be done in a follow up PR to not overload this refactor. Also it's more sinsp modifies state ... so we could rename one more time as well.

Reply:
> we might as well do the refactor of only enforcing the EF_MODIFIES_STATE and drop the UF_NEVER_DROP enforcement

Yep, i'd leave this to another PR since it is a somewhat disruptive change!

>  would love to see scap_get_modifies_state_ppm_sc be defined in this API

You mean a `sinsp_modifies_state_set`? This is what `enforce_sinsp_state_ppm_sc` already does, right?

Reply:
Re your q: `sinsp_state_sc_set()` calls `scap_get_modifies_state_ppm_sc` and not sure why `scap_get_modifies_state_ppm_sc` lives in scap. It appears that we could move all the code directly to `sinsp_state_sc_set()` which may not only simplify everything, but also increase the clarity wrt sinsp <-> scap responsibility separation, something Jason and others brought up too in past threads.

We can chat more about it in the follow up PR :)

Reply:
This would make a lot of sense; BUT we first would need to drop `UF_NEVER_DROP` check, since syscall table is not exposed to sinsp ;) 
We can do it later once we decide dropping `UF_NEVER_DROP` set is something we actually want!

Reply:
Yup you probably already know my opinion on `UF_NEVER_DROP` :upside_down_face: would urge to get rid of it for the syscalls selection to have proper separation.

Reply:
Well, `UF_NEVER_DROP` is not taken into account unless you enable dropping mode ;) but Falco was just cleaned up about that: https://github.com/falcosecurity/falco/pull/2415